neural/predict_with_neural.py

```diff
@@ -0,0 +1,217 @@
+import os
+import pandas as pd
+from pytorch_pretrained_bert import BertTokenizer
+import requests
+import torch
+
+import autodiscern.transformations as adt
+from neural.data_processor import DataDictProcessor
+from neural.dataset import generate_docpartition_per_question
+from neural.model import BertEmbedder, generate_sents_embeds_from_docs
+from neural.run_workflow import predict_neural_discern, get_saved_config, return_attnw_over_sents
+from neural.utilities import create_directory, ReaderWriter, get_device
+from neural.neural_discern_run_script import load_biobert_model
+from typing import Dict, List
+
+
+BASE_DIR = '/opt/data/autodiscern'
+questions = [4, 5, 9, 10, 11]
+
+
+def identify_attended_senteces(docid_attnweights_map, proc_articles_repr, topk=5):
+    attended_sents = return_attnw_over_sents(docid_attnweights_map, proc_articles_repr, topk)
+    return attended_sents
+
+
+def run_predict(q_docpartitions, q_fold_config_map, bertmodel, q_state_dict_path_map, results_dir, sents_embed_dir,
+                to_gpu, gpu_index, num_epochs=1):
+    fold_num = 0
+    q_predictions = {}
+    for question in q_fold_config_map:
+        mconfig, options, __ = q_fold_config_map[question]
+        options['num_epochs'] = num_epochs  # override number of epochs using user specified value
+
+        # update options fold num to the current fold
+        options['fold_num'] = fold_num
+        data_partition = q_docpartitions[question][fold_num]
+
+        results_path = os.path.join(results_dir, 'question_{}'.format(question), 'fold_{}'.format(fold_num))
+        results_wrk_dir = create_directory(results_path)
+
+        q_predictions[question] = predict_neural_discern(data_partition, bertmodel, mconfig, options,
+                                                         results_wrk_dir, sents_embed_dir,
+                                                         state_dict_dir=q_state_dict_path_map[question], to_gpu=to_gpu,
+                                                         gpu_index=gpu_index)
+    return q_predictions
+
+
+def retrieve_page_from_internet(url: str):
+    res = requests.get(url)
+    html_page = res.content.decode("utf-8")
+    return html_page
+
+
+def create_prediction_qdoc_partitions(questions: List[int], fold_num: int):
+    q_docpartitions = {}
+    for q in questions:
+        q_docpartitions[q] = {}
+        q_docpartitions[q][fold_num] = {'train': [], 'validation': [], 'test': [0]}  # output of np.vectorize() ?!
```

Comment:
I think just having `{'test':[0]}` is sufficient for the prediction on new samples. I am assuming `0` is the ID assigned to the new webpage/document.